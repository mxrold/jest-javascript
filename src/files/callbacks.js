export function callbackHell(cb) {
    cb('Hola Mundo')
}