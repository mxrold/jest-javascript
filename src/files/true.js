export const beNull = () => null

export const beUndefined = () => undefined

export const beDefined = () => 'This is a string'

export const beTruthy = () => true

export const beFalsy = () => false

