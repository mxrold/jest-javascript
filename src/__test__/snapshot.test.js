import { getCharacter } from "../files/snapshot"
import rick from '../../rick.json'

describe('Test snapshot', () => {
    test('Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot()
    })

    test('Always failed the snapshot', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20)
        }

        expect(user).toMatchSnapshot()
    })

    test('We have an exception', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: 'Martin Roldan'
        }
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        })
    })
})