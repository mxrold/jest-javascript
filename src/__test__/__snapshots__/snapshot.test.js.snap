// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Always failed the snapshot 1`] = `
Object {
  "createAt": 2021-08-07T15:13:38.675Z,
  "id": 15,
}
`;

exports[`Test snapshot Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanches",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Test snapshot We have an exception 1`] = `
Object {
  "id": Any<Number>,
  "name": "Martin Roldan",
}
`;
